core/src/main/java/org/gbif/pipelines/core/functions/transforms/ExtendedRecordToLocationTransformer.java

```diff
@@ -0,0 +1,187 @@
+    }
+    loc.setContinent(interpretedContinent);
+
+    loc.setWaterBody(record.getCoreTerms().get(DwCATermIdentifier.waterBody.getIdentifier()));
+    loc.setIslandGroup(record.getCoreTerms().get(DwCATermIdentifier.islandGroup.getIdentifier()));
+    loc.setIsland(record.getCoreTerms().get(DwCATermIdentifier.island.getIdentifier()));
+
+    /*
+      Interpreting Country code
+     */
+    CharSequence rawCountry = record.getCoreTerms().get(DwCATermIdentifier.country.getIdentifier());
+    CharSequence rawCountryCode = record.getCoreTerms().get(DwCATermIdentifier.countryCode.getIdentifier());
+    String interpretedCountry = null;
+    String interpretedCountryCode = null;
+    if (rawCountry != null) {
+      List<Issue> issues = new ArrayList<>();
+      List<Lineage> lineages = new ArrayList<>();
+      ParseResult<Country> parseCountry = CountryParser.getInstance().parse(rawCountry.toString().trim());
+      if (parseCountry.isSuccessful()) {
+        interpretedCountry = parseCountry.getPayload().getTitle();
+        interpretedCountryCode = parseCountry.getPayload().getIso3LetterCode();
+      } else {
+        issues.add(Issue.newBuilder()
+                     .setIssueType(IssueType.OTHERS)
+                     .setRemark("Country parse failed")
+                     .build());
+        lineages.add(Lineage.newBuilder()
+                       .setLineageType(LineageType.OTHERS)
+                       .setRemark("Since the parse on country failed, interpreting as null")
+                       .build());
+      }
+      fieldIssueMap.put(DwCATermIdentifier.country.name(), issues);
+      fieldLineageMap.put(DwCATermIdentifier.country.name(), lineages);
+    } else if (rawCountryCode != null) {
+      List<Issue> issues = new ArrayList<>();
+      List<Lineage> lineages = new ArrayList<>();
+      ParseResult<Country> parseCountry = CountryParser.getInstance().parse(rawCountryCode.toString().trim());
+      if (parseCountry.isSuccessful()) {
+        interpretedCountry = parseCountry.getPayload().getTitle();
+        interpretedCountryCode = parseCountry.getPayload().getIso3LetterCode();
+      } else {
+        issues.add(Issue.newBuilder()
+                     .setIssueType(IssueType.OTHERS)
+                     .setRemark(parseCountry.getError().getMessage())
+                     .build());
+        lineages.add(Lineage.newBuilder()
+                       .setLineageType(LineageType.OTHERS)
+                       .setRemark("Since the parse on countryCode failed, interpreting as null")
+                       .build());
+      }
+      fieldIssueMap.put(DwCATermIdentifier.countryCode.name(), issues);
+      fieldLineageMap.put(DwCATermIdentifier.countryCode.name(), lineages);
+    }
+    loc.setCountry(interpretedCountry);
+    loc.setCountryCode(interpretedCountryCode);
+
+    loc.setStateProvince(record.getCoreTerms().get(DwCATermIdentifier.stateProvince.getIdentifier()));
```

Comment:
I'm doubtful if we need these assignations, don't take this comment as request to change the code, but maybe we can change this to be something like COALESCE(interpretedValue, rawValue) further in the pipeline